### PR TITLE
increase transaction timeout

### PR DESCRIPTION
**Pull request by callmehiphop**
Closes https://github.com/GoogleCloudPlatform/google-cloud-node/pull/2644

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -125,11 +125,18 @@ function Transaction(session, options) {
...
+  options = options || {};
+
   this.queuedMutations_ = [];
   this.runFn_ = null;
 
-  this.timeout_ = DEFAULT_TRANSACTION_TIMEOUT;
   this.beginTime_ = null;
+  this.timeout_ = DEFAULT_TRANSACTION_TIMEOUT;
+
+  if (options.timeout) {
```
Comment (stephenplusplus) [inline src/transaction.js]: Will anyone ever want `0`?

File: src/transaction.js [hunk 2]
```diff
@@ -125,11 +125,18 @@ function Transaction(session, options) {
...
+    this.timeout_ = options.timeout;
+    delete options.timeout;
```
Comment (stephenplusplus) [inline src/transaction.js]: We should probably not delete the user-held property (unless `options` is a cloned object already?)

File: src/transaction.js (hunk 1 above)
Comment (callmehiphop) [reply src/transaction.js]: I don't think so, but I've added a `is.number()` just to be safe.

File: src/transaction.js (hunk 2 above)
Comment (callmehiphop) [reply src/transaction.js]: Good call, I've made an update PTAL

**Pull request merged**